sdk/eventhub/Azure.Messaging.EventHubs/src/Diagnostics/EventDataInstrumentation.cs

```diff
@@ -2,20 +2,30 @@
 // Licensed under the MIT License.
 
 using System.Diagnostics;
+using Azure.Core.Pipeline;
 
 namespace Azure.Messaging.EventHubs.Diagnostics
 {
     internal class EventDataInstrumentation
     {
         private const string DiagnosticIdProperty = "Diagnostic-Id";
 
-        public static bool InstrumentEvent(EventData eventData, Activity activity)
+        private static readonly string MessageActivityName = $"{nameof(Azure)}.{nameof(Messaging)}.{nameof(EventHubs)}.Message";
+
+        public static bool InstrumentEvent(ClientDiagnostics clientDiagnostics, EventData eventData)
         {
-            if (activity != null &&
-                !eventData.Properties.ContainsKey(DiagnosticIdProperty))
+            if (!eventData.Properties.ContainsKey(DiagnosticIdProperty))
             {
-                eventData.Properties[DiagnosticIdProperty] = activity.Id;
-                return true;
+                using DiagnosticScope messageScope = clientDiagnostics.CreateScope(MessageActivityName);
+                messageScope.AddAttribute("kind", "internal");
```

Comment:
i think it's not needed, internal  default in opentelemetry and is a good default in general case